### PR TITLE
Mohamed user cell pages

### LINKED ISSUE #16: Third issue: display cell phone on user profile page
Part of #6 

On the /profile page (i.e. in the frontend/src/main/pages/ProfilePage.js component) the cell phone number is displayed.

### PR DESCRIPTION
**Pull request by Mohamed-Elfouly**
Closes #16 #17 #6 

As a user, you are able to change your phone number via the interface on the profile page. Shown below. Input is taken via a modal also shown below. Change is shown instantly without the need to refresh the page. 100% mutation tested.

image

image

### CODE REVIEW

**Comment by Mohamed-Elfouly** [comment]
image

**Comment by cjyu81** [comment]
good to merge

**Review by cjyu81 — APPROVED** [review]
Looks good. Approving

File: frontend/src/main/pages/ProfilePage.js [hunk 1]
```diff
@@ -3,18 +3,76 @@ import { Row, Col } from "react-bootstrap";
...
 
 const ProfilePage = () => {
+    
 
     const { data: currentUser } = useCurrentUser();
+    const [show, setShow] = useState(false);
+    //Stryker disable all
+    const [phoneNumber, setPhoneNumber] = useState("");
+    const [updatedPhoneNumber, setUpdatedPhoneNumber] = useState("");
+    //striker enable all
```
Comment (pconrad) [inline frontend/src/main/pages/ProfilePage.js:22]:
```suggestion
    // Stryker restore all
```
This is the proper version of the comment that restores Stryker testing.

File: frontend/src/main/pages/ProfilePage.js [hunk 2]
```diff
@@ -28,13 +86,51 @@ const ProfilePage = () => {
...
+          <Form onSubmit={onSubmit}>
+            <Form.Group className="mb-3">
+              <Form.Label>Phone number</Form.Label>
+              <Form.Control
+                type="text"
+                id="PhoneInput"
+                data-testid="PhoneInput"
+                placeholder="###-###-####"
+                autoFocus
+                onChange={onChangePhoneNumber} // Update phone number state
```
Comment (pconrad) [inline frontend/src/main/pages/ProfilePage.js:117]:
```suggestion
                onChange={onChangePhoneNumber}
```

I wonder whether the comment is adding any valuable information here, or is just clutter?

File: frontend/src/tests/pages/CellPhone.test.js [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+
+        const saveButton = screen.getByText("Save Changes");
+        fireEvent.click(saveButton);
+        
+        await waitFor(() => expect(screen.queryByText("Input phone number")).not.toBeInTheDocument());
+        await waitFor(() => expect(axiosMock.history.put.length).toBe(1));
+        
+        //"Required request parameter 'cellPhone' for method parameter type String is not present"
+        await waitFor(() => expect(mockToast).toBeCalledWith("Cell Phone number changed 987-654-3210"));
+        //await waitFor(() => expect(screen.getByText("987-654-3210")).toBeInTheDocument());
```
Comment (pconrad) [inline frontend/src/tests/pages/CellPhone.test.js:156]: We try not to commit commented out code into the main branch

```suggestion
```

File: frontend/src/tests/pages/CellPhone.test.js [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+        const phoneInput = screen.getByPlaceholderText("###-###-####");
+        fireEvent.change(phoneInput, { target: { value: '987-654-3210' } });
+
+        const saveButton = screen.getByText("Save Changes");
+        fireEvent.click(saveButton);
+        
+        await waitFor(() => expect(screen.queryByText("Input phone number")).not.toBeInTheDocument());
+        await waitFor(() => expect(axiosMock.history.put.length).toBe(1));
+        
+        //"Required request parameter 'cellPhone' for method parameter type String is not present"
```
Comment (pconrad) [inline frontend/src/tests/pages/CellPhone.test.js:154]: The purpose of this comment is unclear.  Can we remove it?
```suggestion
```

File: frontend/src/tests/pages/CellPhone.test.js [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+
+    test("on cancel", async () => {
+
+        const axiosMock =new AxiosMockAdapter(axios);
+        axiosMock.onGet("/api/currentUser").reply(200, apiCurrentUserFixtures.userOnly);
+        axiosMock.onGet("/api/systemInfo").reply(200, systemInfoFixtures.showingNeither);
+        axiosMock.onPut('/api/userprofile/update-cellPhone').reply(202);
+        
+
+        //axiosMock.onPut("/api/userprofile/update-cellPhone?cellPhone=123-456-7890").reply(202);
```
Comment (pconrad) [inline frontend/src/tests/pages/CellPhone.test.js:171]:
```suggestion
```

Commented out code?

File: frontend/src/tests/pages/CellPhone.test.js [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+        
+        //"Required request parameter 'cellPhone' for method parameter type String is not present"
+        await waitFor(() => expect(mockToast).not.toBeCalledWith("Cell Phone number changed 987-654-3210"));
+        await waitFor(() => expect(screen.queryByText("987-654-3210")).not.toBeInTheDocument());
+
+    });*/
+
+    //TODO: Create test with custom user that starts with a CellPhone, mutation testing then DONE!!!!!
+    
+});
```
Comment (pconrad) [inline frontend/src/tests/pages/CellPhone.test.js:252]:
```suggestion
```

Please remove this huge section of commented out code.

**Review by pconrad — CHANGES_REQUESTED** [review]
There's one correction, and a lot of commented out code to fix.

**Review by cjyu81 — APPROVED** [review]
Looks good

**Review by cjyu81 — DISMISSED** [review]
Looks good. Approved

**Review by pconrad — DISMISSED** [review]
LGTM

**Pull request merged**